### PR TITLE
enhance `Toolchain.get_flag` to handle lists

### PR DESCRIPTION
**Pull request by branfosj**
Alternative to #4318

### CODE REVIEW

**Comment by boegel** [comment]
@branfosj Are you up for implementing a test for `get_flag` in `test/framework/toolchain.py` to confirm this is working as designed now for both string and list inputs? It looks like we don't have any yet...

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/tools/toolchain/toolchain.py [hunk 1]
```diff
@@ -1129,8 +1129,11 @@ def _setenv_variables(self, donotset=None, verbose=True):
             setvar("EBVAR%s" % key, val, verbose=False)
 
     def get_flag(self, name):
-        """Get compiler flag for a certain option."""
-        return "-%s" % self.options.option(name)
+        """Get compiler flag(s) for a certain option."""
+        if isinstance(self.options.option(name), list):
```
[the remark this answers is not in the document]
Comment (branfosj) [reply easybuild/tools/toolchain/toolchain.py:1133]: No. That will require more investigation as I'm getting the following error if I try to add that:
```
TypeError: 'tuple' object does not support item assignment
```

[the remark this answers is not in the document]
Comment (branfosj) [reply easybuild/tools/toolchain/toolchain.py:1133]: The full traceback after I added `or isinstance(self.options.option(name), tuple)` to the above and have `'-foo -bar': ('foo', 'bar'),` in the tests.

```
Traceback (most recent call last):
  File "/home/simon/work/easybuild-framework/test/framework/toolchain.py", line 2969, in test_get_flag
    self.assertEqual(tc.get_flag('openmp'), flagstring)
  File "/home/simon/work/easybuild-framework/easybuild/tools/toolchain/toolchain.py", line 1133, in get_flag
    if isinstance(self.options.option(name), list) or isinstance(self.options.option(name), tuple):
  File "/home/simon/work/easybuild-framework/easybuild/tools/toolchain/options.py", line 118, in option
    res[idx] = elem % templatedict
TypeError: 'tuple' object does not support item assignment
```

[the remark this answers is not in the document]
Comment (branfosj) [reply easybuild/tools/toolchain/toolchain.py:1133]: If I set `self.options.options_map['openmp'] = ('fiopenmp', )` and try a build I get
```
ERROR: Traceback (most recent call last):
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/main.py", line 133, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/framework/easyblock.py", line 4236, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/framework/easyblock.py", line 4119, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/framework/easyblock.py", line 3954, in run_step
    step_method(self)()
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/framework/easyblock.py", line 2667, in prepare_step
    rpath_include_dirs=self.rpath_include_dirs)
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/tools/toolchain/toolchain.py", line 859, in prepare
    self.set_variables()
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/toolchains/iimpi.py", line 108, in set_variables
    IntelCompilersToolchain.set_variables(self)
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/toolchains/compiler/intel_compilers.py", line 123, in set_variables
    Compiler.set_variables(self)
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/tools/toolchain/compiler.py", line 174, in set_variables
    self._set_compiler_flags()
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/tools/toolchain/compiler.py", line 250, in _set_compiler_flags
    flags = [self.options.option(x) for x in self.COMPILER_FLAGS if self.options.get(x, False)]
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/tools/toolchain/compiler.py", line 250, in <listcomp>
    flags = [self.options.option(x) for x in self.COMPILER_FLAGS if self.options.get(x, False)]
  File "/rds/projects/2017/branfosj-rse/easybuild/src/easybuild-framework/easybuild/tools/toolchain/options.py", line 115, in option
    res[idx] = elem % templatedict
TypeError: 'tuple' object does not support item assignment
```

That is before I get to `get_flag`. So there is further work to be done to support tuples at all. I consider that out of scope for this PR.

[the remark this answers is not in the document]
Comment (ocaisa) [reply easybuild/tools/toolchain/toolchain.py:1133]: Ok, agreed

**Review by ocaisa — APPROVED** [review]
LGTM

**Pull request merged**